src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts

```diff
@@ -100,6 +103,26 @@ export class TerminalEditorInput extends EditorInput {
 		return false;
 	}
 
+	override async confirm(): Promise<ConfirmResult> {
```

Comment:
Does this happen once per editor that implements it? If I have multiple terminals will this result in multiple of these dialogs followed by a regular dirty file dialog?

If so is there any way we could avoid this to combine all terminals into a single dialog?

Reply:
Yeah one per terminal, this would not support showing a combined dialog for multiple terminals.